aspnet/RVTR.Lodging.WebApi/Controllers/LodgingController.cs

```diff
@@ -1,110 +1,139 @@
-    ///
-    /// </summary>
-    /// <param name="lodging"></param>
-    /// <returns></returns>
-    [HttpPost]
-    public async Task<IActionResult> Post(LodgingModel lodging)
-    {
-      await _unitOfWork.Lodging.InsertAsync(lodging);
-      await _unitOfWork.CommitAsync();
+        /// <summary>
+        ///
+        /// </summary>
+        /// <returns></returns>
+        [HttpGet]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        public async Task<IActionResult> Get()
+        {
+            var lodgings = await _unitOfWork.Lodging.SelectAsync();
+            if (lodgings == null)
+                return NotFound();
+            else
+                return Ok(lodgings);
+        }
 
-      return Accepted(lodging);
-    }
+        /// <summary>
+        ///
+        /// </summary>
+        /// <param name="id"></param>
+        /// <returns></returns>
+        [HttpGet("{id}")]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        public async Task<IActionResult> Get(int id)
+        {
+            if (id <= 0)
+            {
+                return NotFound(id);
+            }
+            else
+            {
+                try
+                {
+                    return Ok(await _unitOfWork.Lodging.SelectAsync(id));
+                }
+                catch
+                {
+                    return NotFound(id);
+                }
+            }
+        }
 
-    /// <summary>
-    ///
-    /// </summary>
-    /// <param name="lodging"></param>
-    /// <returns></returns>
-    [HttpPut]
-    public async Task<IActionResult> Put(LodgingModel lodging)
-    {
-      _unitOfWork.Lodging.Update(lodging);
-      await _unitOfWork.CommitAsync();
+        /// <summary>
+        ///
+        /// </summary>
+        /// <param name="lodging"></param>
+        /// <returns></returns>
+        [HttpPost]
+        public async Task<IActionResult> Post(LodgingModel lodging)
+        {
+            await _unitOfWork.Lodging.InsertAsync(lodging);
+            await _unitOfWork.CommitAsync();
+
+            return Accepted(lodging);
+        }
+
+        /// <summary>
+        ///
+        /// </summary>
+        /// <param name="lodging"></param>
+        /// <returns></returns>
+        [HttpPut]
+        public async Task<IActionResult> Put(LodgingModel lodging)
+        {
+            _unitOfWork.Lodging.Update(lodging);
+            await _unitOfWork.CommitAsync();
+
+            return Accepted(lodging);
+        }
 
-      return Accepted(lodging);
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <returns></returns>
+        [HttpGet]
+        [Route("getAvailableLodgings")]
+        public async Task<IActionResult> GetAvailableLodgings()
+        {
+            return Ok(await _unitOfWork.Lodging.AvailableLodgings());
+        }
     }
-  }
 }
```

Comment:
Might be missing a bracket here, build and see if there are errors

Reply:
No brackets missing.